### PR TITLE
Feature/user stories section one

### PR DESCRIPTION
**Pull request by sme93**
# Pull Request Template
## Description
Please include a summary of the change and which issue is fixed. Simplified populate cards. Made toggle of home and view favorites work. Refactored User to add in pantry property.

Please include which files need to be reviewed. User.js, scripts.js

Fixes # (issue)
## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/network-requests.js [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+    .catch(err => console.log('ERROR', err))
 
-  let ingredientData = fetch('http://localhost:3001/api/v1/ingredients')
+  const ingredientData = fetch('http://localhost:3001/api/v1/ingredients')
     .then(response => response.json())
     .then(ingredientData => {
       return ingredientData;
     })
-    .catch(err => console.log('ERROR'))
+    .catch(err => console.log('ERROR', err))
```
Comment (sarahlane8) [inline src/network-requests.js:14]: Thanks for adding this!

File: src/network-requests.js [hunk 2]
```diff
@@ -1,28 +1,28 @@
 const getData = () => {
-  let userData = fetch('http://localhost:3001/api/v1/users')
+  const userData = fetch('http://localhost:3001/api/v1/users')
```
Comment (sarahlane8) [inline src/network-requests.js:2]: Thanks for changing all these to const!

File: src/scripts.js [hunk 3]
```diff
@@ -13,65 +9,42 @@ import User from './user';
...
   getData()
     .then(allData => {
-      let userId = (Math.floor(Math.random() * allData.userData.length) + 1)
-      let newUser = allData.userData.find(user => {
-        return user.id === Number(userId);
-      });
-      user = new User(userId, newUser.name, newUser.pantry)
-      pantry = new Pantry(newUser.pantry)
-      let cookbook = new Cookbook(allData.recipeData);
+      const randomIndexInArray = Math.floor(
```
Comment (sarahlane8) [inline src/scripts.js:25]: I like this refactor! Cleaner. Easier to read and understand.

File: src/scripts.js [hunk 4]
```diff
@@ -13,65 +9,42 @@ import User from './user';
...
-      <button id='${recipe.id}' aria-label='favorite-button' class='favorite favorite-active card-button'>
-      </button></header>
-      <span id='${recipe.id}' class='recipe-name'>${recipe.name}</span>
-      <img id='${recipe.id}' tabindex='0' class='card-picture'
-      src='${recipe.image}' alt='Food from recipe'>
-      </div>`)
-        })
-      }
-    })
+function viewFavorites(event) {
```
Comment (sarahlane8) [inline src/scripts.js:36]: Holy moly this looks so much better, much more intuitive.

File: src/scripts.js [hunk 5]
```diff
@@ -81,6 +54,7 @@ function greetUser() {
 }
 
 function favoriteCard(event) {
+  const specificRecipe = "";
```
Comment (sarahlane8) [inline src/scripts.js:57]: I think this has to be a let? if we are going to reassign it below?

File: src/scripts.js [hunk 6]
```diff
@@ -101,18 +75,20 @@ function favoriteCard(event) {
...
-        favoriteCard(event);
-      } else if (event.target.classList.contains('card-picture')) {
-        displayDirections(event);
-      } else if (event.target.classList.contains('home')) {
-        favButton.innerHTML = 'View Favorites';
-        let cookbook = new Cookbook(allData.recipeData);
-        populateCards(cookbook.recipes);
-      }
-    })
+  console.log("event ", event);
```
Comment (sarahlane8) [inline src/scripts.js:78]: remove this console.log

File: src/scripts.js [hunk 7]
```diff
@@ -101,18 +75,20 @@ function favoriteCard(event) {
...
+  // getData()
+  //   .then(allData => {
+  if (event.target.classList.contains('favorite')) {
+    favoriteCard(event);
+  } else if (event.target.classList.contains('card-picture')) {
+    displayDirections(event);
+  } else if (event.target.classList.contains('home')) {
+    favButton.innerHTML = 'View Favorites';
+    console.log('fjkdhsfkjsdhf')
```
Comment (sarahlane8) [inline src/scripts.js:87]: remove this console.log

**Review by sarahlane8 — DISMISSED** [review]
Does line 57 need to be a let, instead of const? in scripts

There are 2 console.logs that need to be removed in the scripts file. line 78 and 87

I'll have to see how this all plays out, but we don't need the getFavorites function?

File: src/scripts.js (hunk 7 above)
Comment (sme93) [reply src/scripts.js:87]: work in progress, will remove later of course

File: src/scripts.js (hunk 6 above)
Comment (sme93) [reply src/scripts.js:78]: work in progress, will remove later

File: src/scripts.js (hunk 5 above)
Comment (sme93) [reply src/scripts.js:57]: actually i'm still working through this function, this variable is declared here and I haven't even used it yet - so yeah, work in progress

**Pull request merged**